compiler/frontend/spark/src/main/java/edu/snu/nemo/compiler/frontend/spark/core/SparkFrontendUtils.java

```diff
@@ -132,10 +134,24 @@ public static Serializer deriveSerializerFrom(final org.apache.spark.SparkContex
    * @return the converted Java function.
    */
   public static <I, O> Function<I, O> toJavaFunction(final Function1<I, O> scalaFunction) {
+    final ClassTag<Function1<I, O>> classTag = ClassTag$.MODULE$.apply(scalaFunction.getClass());
```

Comment:
Why these serialization & deserialization are needed? Please update the method comment.

Reply:
Also, why don't we need to use this technique for other function converting methods like [this](https://github.com/apache/incubator-nemo/pull/113/files#diff-1a3fce9011479d4662455c65913ec807R168)?